garage/tf/optimizers/lbfgs_optimizer.py

```diff
@@ -67,39 +68,40 @@ def loss(self, inputs, extra_inputs=None):
             extra_inputs = list()
         return self._opt_fun["f_loss"](*(list(inputs) + list(extra_inputs)))
 
-    def optimize(self, inputs, extra_inputs=None):
-        f_opt = self._opt_fun["f_opt"]
+    def optimize(self, inputs, extra_inputs=None, name=None):
+        with tf.name_scope(name, "optimize", values=[inputs, extra_inputs]):
+            f_opt = self._opt_fun["f_opt"]
 
-        if extra_inputs is None:
-            extra_inputs = list()
+            if extra_inputs is None:
+                extra_inputs = list()
 
-        def f_opt_wrapper(flat_params):
-            self._target.set_param_values(flat_params, trainable=True)
-            ret = f_opt(*inputs)
-            return ret
-
-        itr = [0]
-        start_time = time.time()
-
-        if self._callback:
-
-            def opt_callback(params):
-                loss = self._opt_fun["f_loss"](*(inputs + extra_inputs))
-                elapsed = time.time() - start_time
-                self._callback(
-                    dict(
-                        loss=loss,
-                        params=params,
-                        itr=itr[0],
-                        elapsed=elapsed,
-                    ))
-                itr[0] += 1
-        else:
-            opt_callback = None
-
-        scipy.optimize.fmin_l_bfgs_b(
-            func=f_opt_wrapper,
-            x0=self._target.get_param_values(trainable=True),
-            maxiter=self._max_opt_itr,
-            callback=opt_callback,
-        )
+            def f_opt_wrapper(flat_params):
```

Comment:
do we not need name_scopes here?

Reply:
Inside f_opt_wrapper there's two calls, one for set_param_values and one for f_opt. The former is defined [here](https://github.com/rlworkgroup/garage/blob/efb8e142206754582e4c83ec130024020e16a21e/garage/tf/core/parameterized.py#L69) and it already has a scope, while the latter is a TF compiled function that does not generate any variables or operations, so I found redundant to add another scope besides "optimize" to enclose f_opt_wrapper.